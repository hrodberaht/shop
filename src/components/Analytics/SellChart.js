import React, { Component } from 'react';
import PropTypes from 'prop-types';
import sellChart from '../../shared/createSellChart';

export default class SellChart extends Component {
  componentDidMount() {
    sellChart(this.props.data);
  }

  render() {
    return <div id="sell-chart" />;
  }
}

SellChart.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
};
