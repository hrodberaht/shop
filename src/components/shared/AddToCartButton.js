import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCartPlus } from '@fortawesome/free-solid-svg-icons';

export default class AddToCartButton extends Component {
  static propTypes = {
    product: PropTypes.shape({
      name: PropTypes.string,
      type: PropTypes.string,
      price: PropTypes.number,
    }).isRequired,
    handleClickToCart: PropTypes.func.isRequired,
    className: PropTypes.string.isRequired,
  };

  render() {
    const { className, handleClickToCart, product } = this.props;
    return (
      <button type="button" className={className} onClick={() => handleClickToCart(product)}>
        <FontAwesomeIcon icon={faCartPlus} />
      </button>
    );
  }
}
