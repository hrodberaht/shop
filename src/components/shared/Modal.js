import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';

export default class Modal extends Component {
  static propTypes = {
    handleCancelClick: PropTypes.func.isRequired,
    handleConfirmClick: PropTypes.func.isRequired,
  };

  render() {
    const { handleConfirmClick, handleCancelClick } = this.props;
    return ReactDOM.createPortal(
      <div className="modal">
        <div className="inner-modal">
          <h2>Are you sure?</h2>
          <div className="inner-modal__buttons">
            <button type="button" className="btn btn-submit" onClick={handleCancelClick}>
              Cancel
            </button>
            <button type="button" className="btn btn-danger" onClick={handleConfirmClick}>
              Confirm
            </button>
          </div>
        </div>
      </div>,
      document.body,
    );
  }
}
