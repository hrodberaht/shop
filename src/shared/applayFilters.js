const applayFiletrs = (data, filterValues) => {
  if (Object.keys(filterValues).length === 0) {
    return data;
  }
  const regexp = new RegExp(filterValues.id.replace(/[/\\^$*+?.()|[\]{}]/g, '\\$&'), 'i');
  const filteredByStatus = filterValues.status !== 'all'
    ? data.filter(order => order.status === filterValues.status)
    : data;
  return filterValues.id !== ''
    ? filteredByStatus.filter(order => regexp.test(order.id))
    : filteredByStatus;
};
export default applayFiletrs;
