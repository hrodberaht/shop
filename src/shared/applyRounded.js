const applyRounded = price => price.toFixed(2);
export default applyRounded;
