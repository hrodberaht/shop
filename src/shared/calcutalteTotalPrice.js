const calculateTotalPrice = (pcsOrder, price) => pcsOrder * price;

export default calculateTotalPrice;
