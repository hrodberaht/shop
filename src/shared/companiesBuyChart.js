import Highcharts from 'highcharts';
import Exporting from 'highcharts/modules/exporting';

Exporting(Highcharts);

const companiesBuyChart = (data) => {
  const companies = [...data.orders].filter(order => order.companyId !== '');
  const series = [];
  companies.forEach((company) => {
    if (series.length === 0) return series.push(company);
    const index = series.findIndex(serie => serie.companyId === company.companyId);
    // eslint-disable-next-line no-return-assign
    return index > 0
      ? (series[index] = { ...company, totalPrice: series[index].totalPrice + company.totalPrice })
      : series.push(company);
  });
  Highcharts.chart('companies-buy-chart', {
    chart: {
      type: 'bar',
    },
    title: {
      text: 'Sells',
    },
    xAxis: {
      categories: ['Sumary'],
    },
    yAxis: {
      title: {
        text: 'Sells',
      },
    },
    series: series.map(company => ({ name: company.companyId, data: [company.totalPrice] })),
  });
};

export default companiesBuyChart;
