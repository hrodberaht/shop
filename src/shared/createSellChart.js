import Highcharts from 'highcharts';
import Exporting from 'highcharts/modules/exporting';

Exporting(Highcharts);

const sellChart = data => Highcharts.chart('sell-chart', {
  chart: {
    type: 'pie',
  },
  title: {
    text: 'Sold products',
  },
  series: [
    {
      name: 'pcs',
      colorByPoint: true,
      data,
    },
  ],
});

export default sellChart;
