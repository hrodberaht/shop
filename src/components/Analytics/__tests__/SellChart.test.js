import SellChart from '../SellChart';
import * as chart from '../../../shared/createSellChart';

jest.mock('../../../shared/createSellChart');

const defaultProps = {
  data: [],
};

const setup = buildSetup(SellChart, defaultProps);

describe('<SellChart />', () => {
  it('should render without crash', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should call createSellChart', () => {
    const {
      props: { data },
    } = setup();
    const spy = jest.spyOn(chart, 'default');
    expect(spy).toHaveBeenCalledWith(data);
  });
});
