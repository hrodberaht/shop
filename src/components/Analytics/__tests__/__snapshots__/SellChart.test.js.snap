// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SellChart /> should render without crash 1`] = `
<div
  id="sell-chart"
/>
`;
